plugins/inputs/mongodb/mongodb.go

```diff
@@ -135,6 +135,7 @@ func (m *MongoDB) Init() error {
 			opts.ReadPreference = readpref.Nearest()
 		}
 
+		// TODO: should not connect in Init() function!
```

Comment:
connecting in init is still allowed, but it still behaves as badly as it did before. Previously failing to connect in init would stop telegraf from loading. Now failing to connect in init stops the plugin from loading. The biggest change is that there are now tests that check that plugins can init with default config without errors, which wasn't previously checked.

Reply:
What is telegraf going to do with a plugin that failed to init? Just ignore it? A user who is only using toml config and isn't using the new rest api to change config at runtime won't want an init error to be ignored.

A test that checks that each plugin can Init() with default config and doesn't return an error is a great idea. Where is it? I thought it might be in config/config_test.go but I didn't see it.

Reply:
config/testdata/telegraf-agent.toml contains a list of plugins with the non-standard or complicated configs, and `TestAgent_LoadPlugin` tests that it can load it. It's not exhaustive, but we can add to the list or copy this pattern to make a test that tries every plugin. A few need to remove connect from the Init still.